src/main/java/com/iexec/worker/compute/pre/PreComputeService.java

```diff
@@ -81,6 +84,30 @@ public boolean runStandardPreCompute(TaskDescription taskDescription) {
 
     public String runTeePreCompute(TaskDescription taskDescription, WorkerpoolAuthorization workerpoolAuth) {
         String chainTaskId = taskDescription.getChainTaskId();
+        // verify enclave configuration for compute stage
+        TeeEnclaveConfiguration enclaveConfig = taskDescription.getAppEnclaveConfiguration();
+        if (StringUtils.isEmpty(enclaveConfig.getFingerprint())
+                || BytesUtils.stringToBytes(enclaveConfig.getFingerprint()).length != 32) {
+            log.error("Enclave configuration should define a proper " +
+                            "fingerprint [chainTaskId:{}, fingerprint:{}]",
+                    chainTaskId, enclaveConfig.getFingerprint());
+            return "";
+        }
+        long teeComputeMaxHeapSize = DataSize
+                .ofGigabytes(workerConfigService.getTeeComputeMaxHeapSizeGB())
+                .toBytes();
+        if (enclaveConfig.getHeapSize() <= 0
+                || enclaveConfig.getHeapSize() > teeComputeMaxHeapSize) {
+            log.error("Enclave configuration should define a proper heap " +
+                            "size [chainTaskId:{}, heapSize:{}]",
```

Comment:
Just for convenience, we can print the max heap size.

Reply:
Also, it's better to print heap sizes in Mb (or Gb) instead of bytes.

Reply:
Max heap size printed :+1: